src/inhibitors/requiredSettings.js

```diff
@@ -3,10 +3,19 @@ const { Inhibitor } = require('klasa');
 module.exports = class extends Inhibitor {
 
 	async run(message, command) {
-		if (!command.requiredSettings.length || message.channel.type !== 'text') return;
-		// eslint-disable-next-line eqeqeq
-		const requiredSettings = command.requiredSettings.filter(setting => message.guild.settings.get(setting) == null);
-		if (requiredSettings.length) throw message.language.get('INHIBITOR_REQUIRED_SETTINGS', requiredSettings);
+		const gateways = Object.keys(command.requiredSettings);
+		if (!gateways.length) return;
+
+		const missingSettings = [];
+
+		for (const gateway of gateways) {
```

Comment:
Can't this use a for-of loop instead?

Reply:
doesn't it already?

Reply:
Woops, didn't word it right! I meant a for-of with Object.entries from the settings object